### ISSUE TITLE
Implement business logic for program that converts °F to °C

### ISSUE THREAD

**Issue by billhiggins** [ayhiggins/experiments#5]
With this issue, you’ll start to build your own program with a focus on implementing business logic and avoiding common error conditions.

The program is called `fahr_to_cel.py` and converts a temperature specified in degrees Fahrenheit to the corresponding temperature in degrees Celsius. To be more useful, the program allows the user to pass in the source temperature in degrees Fahrenheit via the command line. Here are a few usage examples for invoking the program:

```
# Expected result: 0 (freezing point of water)
python fahr_to_cels.py 32

# Expected result: 100 (boiling point of water)
python fahr_to_cels.py 212

# Expected result: 22 (comfortable room temperature)
python fahr_to_cels.py 72
```

I will write the initial program that captures the passed in Fahrenheit value as a Python variable (`degrees_f`); your job is to write the business logic to convert it to Celsius.

**Acceptance criteria**

- [x] The program correctly converts Fahrenheit temperatures to Celsius temperatures
- [ ] The program is robust against both user input errors as well as logical errors

**Comment by billhiggins** [comment]
Reopened this issue because it’s incomplete. The program’s “happy path” works fine, but it is not robust against input and logical errors—in other words you’ve only met one of the two acceptance criteria.

Please work on making the program more robust, thinking about things that could go wrong, and how you can guard against them. Perhaps we could start by you adding a comment below listing things that could go wrong and (without any code) stating how you might guard against them.

**Comment by ayhiggins** [comment]
We can do the same thing that we did for the adder.py where we add a safeguard for if the user does not put an integer in, or if they provide no argument, we can just say, please enter an integer, or the number you entered is not an integer.

**Comment by ayhiggins** [comment]
another thing that could go wrong are like you mentioned, kelvin and degrees of temperature that are too extreme, I can try to think on making a message pop up to the user if they get kelvin, or if they even go below kelvin, "this number is not scientifically possible to reach" or something along those lines

**Comment by billhiggins** [comment]
Based on your comments, we created [a commit](https://github.com/ayhiggins/experiments/commit/fcb0b0e79ebd762984d401b758c90ca8e4ecc589) that guarded against the following error conditions:

**Wrong number of arguments**

We checked that `len(sys.argv)` is equal to `2` since we expect the syntax:

`python fahr_to_cel.py 55`

… were `sys.argv[0]` is (as always) the program name, and `sys.argv[1]` is the user’s specified degrees Fahrenheit.

**Non-integer argument**

We ensured that `sys.argv[1]` is a valid integer and had to take care not to flag a negative number as a string using the `lstrip('-')` function.

**Number below absolute zero**

Finally, we guarded against the logic error of a number below absolute zero, or -459.67°F.

**Comment by billhiggins** [comment]
What we learned.

**Writing business logic**

Given the assignment, you performed a Google search to remember the exact formula for converting a temperature in degrees Fahrenheit to a corresponding temperature in degrees Celsius.

After some prompting, you remembered that we should not allow an input temperature below absolute zero (-459.67°F), since that doesn’t make logical sense.

**Guarding against bad input**

Once again we saw that the business logic is a rather small part of the program (three lines of code) while we have to take significant care to guard against bad input.

As you’ll find, a significant part of creating good software is anticipating what might go wrong and guarding against these conditions. We call this making software more _robust_. Another important topic is _resilience_ but we’ll talk about that later.
